pkg/rtc/mediatrack.go

```diff
@@ -95,14 +85,22 @@ type MediaTrackParams struct {
 
 func NewMediaTrack(track *webrtc.TrackRemote, params MediaTrackParams) *MediaTrack {
 	t := &MediaTrack{
-		params:               params,
-		ssrc:                 track.SSRC(),
-		streamID:             track.StreamID(),
-		codec:                track.Codec(),
-		connectionStats:      connectionquality.NewConnectionStats(),
-		done:                 make(chan struct{}),
-		maxSubscriberQuality: make(map[livekit.ParticipantID]livekit.VideoQuality),
-	}
+		params:          params,
+		ssrc:            track.SSRC(),
+		streamID:        track.StreamID(),
+		codec:           track.Codec(),
+		connectionStats: connectionquality.NewConnectionStats(),
+		done:            make(chan struct{}),
+	}
+
+	t.MediaTrackSubscriptions = NewMediaTrackSubscriptions(MediaTrackSubscriptionsParams{
+		MediaTrack:       t,
+		BufferFactory:    params.BufferFactory,
+		ReceiverConfig:   params.ReceiverConfig,
+		SubscriberConfig: params.SubscriberConfig,
+		Telemetry:        params.Telemetry,
+		Logger:           &params.Logger,
```

Comment:
Passing a pointer to logger here as cloud does not have logger yet :-(

Reply:
```suggestion
		Logger:           params.Logger,
```

Logger is an interface backed by a pointer type, we can pass as is.